### DISCUSSION TITLE
第4节课作业(2班)

### DISCUSSION THREAD

**Discussion by MING-ZCH** [InternLM/Tutorial#245] [作业]
### 基础作业

构建数据集，使用 XTuner 微调 InternLM-Chat-7B 模型, 让模型学习到它是你的智能小助手，效果如下图所示，本作业训练出来的模型的输出需要**将不要葱姜蒜大佬**替换成自己名字或昵称！

**微调前**（回答比较官方）
![image](https://github.com/InternLM/tutorial/assets/119648793/f19d0ed2-2d43-4bdf-9b1e-7a89ee9efa18)

**微调后**（对自己的身份有了清晰的认知）
![image](https://github.com/InternLM/tutorial/assets/119648793/4086e12d-a78e-4feb-8294-20eb59c1cf3c)

作业参考答案：https://github.com/InternLM/tutorial/blob/main/xtuner/self.md

**进阶作业：**

- 将训练好的Adapter模型权重上传到 OpenXLab、Hugging Face 或者 MoelScope 任一平台。
- 将训练好后的模型应用部署到 OpenXLab 平台，参考部署文档请访问：https://aicarrier.feishu.cn/docx/MQH6dygcKolG37x0ekcc4oZhnCe

**Comment by songyuc** [comment]
### 基础作业
构建数据集，使用 XTuner 微调 InternLM-Chat-7B 模型, 让模型学习到它是你的智能小助手。

**微调前**（回答比较官方）  
<img width="1245" alt="屏幕截图 2024-01-11 011118" src="https://github.com/InternLM/tutorial/assets/27288110/bb924ab7-7b26-4b58-8a47-f0e51166288c">

**微调后**（对自己的身份有了清晰的认知）
![微信截图_20240111025712](https://github.com/InternLM/tutorial/assets/27288110/28eabf92-8f56-46c0-bf4c-41f9ee5d2b45)

**Comment by YanxingLiu** [comment]
基础作业
构建数据集，使用 XTuner 微调 InternLM-Chat-7B 模型, 让模型学习到它是你的智能小助手。

微调前（回答比较官方）
![image](https://github.com/InternLM/tutorial/assets/42299757/9e6e21e2-7636-42ce-b65e-1e51b314faaf)


微调后（对自己的身份有了清晰的认知）
<img width="838" alt="image" src="https://github.com/InternLM/tutorial/assets/42299757/64db2881-016b-48a0-b2e8-222b0ddb84bb">

**Comment by pranerd** [comment]
基础作业：
微调下的输出：
![xtuner_me](https://github.com/InternLM/tutorial/assets/19358928/dce83206-b836-4a61-b25b-bbc087762133)

**Comment by xulinui** [comment]
https://h90ag9106t.feishu.cn/docx/YjQPdpEJpohTe8xF2UucCguanbe

**Comment by zyy0831** [comment]
原计划训练为3次，但训练第1次后，损失已经很低，且由于学习实践整体流程，一定过拟合，无需消耗第2和3次时间。
![4](https://github.com/InternLM/tutorial/assets/43286447/aeaba185-68d4-40eb-8945-59745ffe3b13)

结果如下图
![image](https://github.com/InternLM/tutorial/assets/43286447/e4c8c194-e4c1-4b7e-95c7-ccb8b6014c97)
![image](https://github.com/InternLM/tutorial/assets/43286447/7871f76d-dd66-4499-9443-05db3bc1c296)
模型运行占用资源情况
![image](https://github.com/InternLM/tutorial/assets/43286447/c10bdff5-1bfe-44f7-a9ae-335d70689bec)
总之，是不错的应用经验

**Comment by DarthThomas** [comment]
### 作业
使用脚本处理python官方doc txt版本的FAQ，生成json，一起加入finetune
![截屏2024-01-14 15 31 40](https://github.com/InternLM/tutorial/assets/20266717/2fec4318-bf8d-4904-b607-26b3ef09ad64)
![截屏2024-01-14 20 23 51](https://github.com/InternLM/tutorial/assets/20266717/b8bc67ad-87da-47f1-925b-631c6a4fe1b8)

**Comment by ZixinxinWang** [comment]
1.
![image](https://github.com/InternLM/tutorial/assets/152623177/58fccb3a-eb17-4ff4-a4ce-25d1a2524989)
2.
<img width="1240" alt="image" src="https://github.com/InternLM/tutorial/assets/152623177/6c900ce0-424c-4d1d-862b-c09872e3228c">

**Comment by CoderJackZhu** [comment]
基础作业
微调前（回答比较官方）
![image](https://github.com/InternLM/tutorial/assets/42195616/fb6c43d1-5010-4aa6-ab6a-f1b764280fc5)

微调后（对自己的身份有了清晰的认知）
![image](https://github.com/InternLM/tutorial/assets/42195616/8c5741ec-f4b0-4fca-ada6-41d9d7e61745)

**Comment by kyrie2to11** [comment]
[InterLM L-4: XTuner 大模型单卡低成本微调](https://zhuanlan.zhihu.com/p/677790607) 基础作业打卡~

**Comment by ljqiff** [comment]
基础作业
3个epch幻觉严重，过拟合，污染原始权重
![image](https://github.com/InternLM/tutorial/assets/29425919/251f0e1c-3ec5-4f5e-8a7e-79657faca4f3)
进阶作业：
![1705247217093](https://github.com/InternLM/tutorial/assets/29425919/ba887e0e-c668-42de-8bf2-6b8c273cf928)
MoelScope 平台。
https://www.modelscope.cn/models/ljqiff/learn4class/summary
OpenXlab
https://openxlab.org.cn/apps/detail/ljqiff/CarLLMTest

**Comment by kaoyansoft123** [comment]
微调前：
<img width="589" alt="045d6db1f65f6772897c960aaea5573" src="https://github.com/InternLM/tutorial/assets/152302734/3dd5ac2a-c190-4eab-8e5b-128403baa504">
微调后：
![image](https://github.com/InternLM/tutorial/assets/152302734/7746786f-5f13-464d-b788-1e0dbddc2646)

**Comment by RyanWu31** [comment]
https://github.com/RyanWu31/tutorial_notebook/blob/main/README.md

**Comment by Mr-Vincenyang** [comment]
[作业](https://zhuanlan.zhihu.com/p/678719352)

**Comment by PrinceRay7** [comment]
做了点有趣的东西
作业：构建数据集，使用 XTuner 微调 InternLM-Chat-7B 模型, 让模型学习到他是一个某公司的招聘HR
Adapter权重已经发布至openXlab
小助手的回答均为虚构，如有雷同，纯属巧合，请勿对号入座
https://github.com/PrinceRay7/InternLM_HW/blob/main/hw4/homework/homework.md

**Comment by kishiokon** [comment]
![4M8W1RE1$ VGRQS3242J~ U](https://github.com/InternLM/tutorial/assets/92700495/634d85b2-7512-4e80-a7d4-c2e48bc13181)
微调之后变成了复读小助手

**Comment by ZixinxinWang** [comment]
![image](https://github.com/InternLM/tutorial/assets/152623177/ad892428-6ea7-4708-a861-40d166e7fad0)
![image](https://github.com/InternLM/tutorial/assets/152623177/9a6e69ec-b23d-4358-9a1a-7ce0f9490037)
补充进阶作业：
权重上传
![image](https://github.com/InternLM/tutorial/assets/152623177/697ccd51-efff-4c0f-8ba8-74e7c3b8d094)
应用部署
![image](https://github.com/InternLM/tutorial/assets/152623177/55da22df-155a-4aeb-b24a-4d21d380237e)
